packages/core/src/combo-box/ComboBox.tsx

```diff
@@ -248,7 +249,11 @@ export const ComboBox = forwardRef(function ComboBox<Item>(
 
         break;
       case "Tab":
-        if (!multiselect && activeState) {
+        if (
+          tabToSelect &&
+          activeState &&
+          !selectedState.includes(activeState?.value)
```

Comment:
This is not captured in test i guess? (not deselect)

Reply:
@origami-z  have added the test for the same